app/controllers/register/establishers/individual/AddressResultsController.scala

```diff
@@ -0,0 +1,94 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.register.establishers.individual
+
+import javax.inject.Inject
+
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import connectors.DataCacheConnector
+import controllers.actions._
+import config.FrontendAppConfig
+import forms.register.establishers.individual.AddressResultsFormProvider
+import identifiers.register.establishers.individual.{AddressId, AddressResultsId, EstablisherDetailsId}
+import models.addresslookup.{Address, AddressRecord}
+import models.requests.DataRequest
+import utils.{Enumerable, MapFormats, Navigator, UserAnswers}
+import views.html.register.establishers.individual.addressResults
+import models.{Index, Mode}
+import play.api.mvc.{Action, AnyContent, Result}
+
+import scala.concurrent.Future
+
+class AddressResultsController @Inject()(
+                                        appConfig: FrontendAppConfig,
+                                        override val messagesApi: MessagesApi,
+                                        dataCacheConnector: DataCacheConnector,
+                                        navigator: Navigator,
+                                        authenticate: AuthAction,
+                                        getData: DataRetrievalAction,
+                                        requireData: DataRequiredAction,
+                                        formProvider: AddressResultsFormProvider) extends FrontendController with I18nSupport
+  with Enumerable.Implicits with MapFormats{
+
+  val form = formProvider()
+
+  def onPageLoad(mode: Mode, index: Index): Action[AnyContent] = (authenticate andThen getData andThen requireData).async {
+    implicit request =>
+      retrieveEstablisherName(index) {
+        establisherName =>
+
+          val result = request.userAnswers.get[Seq[Address]](AddressId) match {
+            case None => Redirect(controllers.register.establishers.individual.routes.AddressController.onPageLoad(mode, index))
+            case Some(value) => Ok(addressResults(appConfig, form, mode, index, value, establisherName))
+          }
+          Future.successful(result)
+      }
+  }
+
+  def onSubmit(mode: Mode, index: Index): Action[AnyContent] = (authenticate andThen getData andThen requireData).async {
+    implicit request =>
+      retrieveEstablisherName(index) {
+        establisherName =>
+          form.bindFromRequest().fold(
+            (formWithErrors: Form[_]) =>
+              Future.successful(BadRequest(addressResults(appConfig, formWithErrors, mode, index,
+                request.userAnswers.get[Seq[Address]](AddressId.path).getOrElse(Seq.empty), establisherName))),
+            (value) =>
+              dataCacheConnector.save(
+                request.externalId,
+                AddressResultsId(index),
```

Comment:
We should just save the address straight into the address form, this means that the address entry page is just a normal page with multiple ways of populating it.